bug: `ClassNotFoundException` after adding `@capacitor-firebase/storage`

### Plugin(s)

- [ ] Analytics
- [ ] App
- [ ] App Check
- [ ] Authentication
- [ ] Crashlytics
- [ ] Cloud Firestore
- [ ] Cloud Messaging
- [ ] Performance
- [ ] Remote Config

### Did you test the latest version?

- [X] I use the latest version

### Platform(s)

- [X] Android
- [ ] iOS
- [ ] Web

### Current behavior

The following error message is thrown when starting the app after adding the `@capacitor-firebase/storage` package:

```
Process: dev.robingenz.capacitorjs.demo.firebaseplugins, PID: 5583
java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/appcheck/interop/InternalAppCheckTokenProvider;
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:536)
	at java.lang.Class.forName(Class.java:467)
	at com.google.firebase.components.ComponentDiscovery.instantiate(ComponentDiscovery.java:120)
	at com.google.firebase.components.ComponentDiscovery.lambda$discoverLazy$0(ComponentDiscovery.java:112)
	at com.google.firebase.components.ComponentDiscovery$$ExternalSyntheticLambda0.get(Unknown Source:2)
	at com.google.firebase.components.ComponentRuntime.discoverComponents(ComponentRuntime.java:115)
	at com.google.firebase.components.ComponentRuntime.<init>(ComponentRuntime.java:99)
	at com.google.firebase.components.ComponentRuntime.<init>(ComponentRuntime.java:46)
	at com.google.firebase.components.ComponentRuntime$Builder.build(ComponentRuntime.java:387)
	at com.google.firebase.FirebaseApp.<init>(FirebaseApp.java:438)
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:295)
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:263)
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:248)
	at com.google.firebase.provider.FirebaseInitProvider.onCreate(FirebaseInitProvider.java:69)
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2451)
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2421)
	at com.google.firebase.provider.FirebaseInitProvider.attachInfo(FirebaseInitProvider.java:61)
	at android.app.ActivityThread.installProvider(ActivityThread.java:7508)
	at android.app.ActivityThread.installContentProviders(ActivityThread.java:7019)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6790)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2132)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.appcheck.interop.InternalAppCheckTokenProvider" on path: DexPathList[[zip file "/data/app/~~Y7goHShL02ZAa9SAyvr3Lg==/dev.robingenz.capacitorjs.demo.firebaseplugins-RUCy_-HDi6dQGzSwmQIuLQ==/base.apk"],nativeLibraryDirectories=[/data/app/~~Y7goHShL02ZAa9SAyvr3Lg==/dev.robingenz.capacitorjs.demo.firebaseplugins-RUCy_-HDi6dQGzSwmQIuLQ==/lib/arm64, /system/lib64, /system_ext/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	at java.lang.Class.classForName(Native Method) 
	at java.lang.Class.forName(Class.java:536) 
	at java.lang.Class.forName(Class.java:467) 
	at com.google.firebase.components.ComponentDiscovery.instantiate(ComponentDiscovery.java:120) 
	at com.google.firebase.components.ComponentDiscovery.lambda$discoverLazy$0(ComponentDiscovery.java:112) 
	at com.google.firebase.components.ComponentDiscovery$$ExternalSyntheticLambda0.get(Unknown Source:2) 
	at com.google.firebase.components.ComponentRuntime.discoverComponents(ComponentRuntime.java:115) 
	at com.google.firebase.components.ComponentRuntime.<init>(ComponentRuntime.java:99) 
	at com.google.firebase.components.ComponentRuntime.<init>(ComponentRuntime.java:46) 
	at com.google.firebase.components.ComponentRuntime$Builder.build(ComponentRuntime.java:387) 
	at com.google.firebase.FirebaseApp.<init>(FirebaseApp.java:438) 
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:295) 
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:263) 
	at com.google.firebase.FirebaseApp.initializeApp(FirebaseApp.java:248) 
	at com.google.firebase.provider.FirebaseInitProvider.onCreate(FirebaseInitProvider.java:69) 
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2451) 
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2421) 
	at com.google.firebase.provider.FirebaseInitProvider.attachInfo(FirebaseInitProvider.java:61) 
	at android.app.ActivityThread.installProvider(ActivityThread.java:7508) 
	at android.app.ActivityThread.installContentProviders(ActivityThread.java:7019) 
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6790) 
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2132) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7918) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
```

### Expected behavior

No error should be thrown.

### Reproduction

https://github.com/robingenz/capacitor-firebase-plugin-demo

### Steps to reproduce

1. Start the app

### Other information

_No response_

### Capacitor doctor

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 5.5.0
  @capacitor/core: 5.5.0
  @capacitor/android: 5.5.0
  @capacitor/ios: 5.5.0

Installed Dependencies:

  @capacitor/cli: 5.4.2
  @capacitor/core: 5.4.2
  @capacitor/ios: 5.4.2
  @capacitor/android: 5.4.2

[success] iOS looking great! 👌
[success] Android looking great! 👌

### Before submitting

- [X] I understand that incomplete issues (e.g. without reproduction) are closed.

--- a/.changeset/sour-goats-scream.md
+++ b/.changeset/sour-goats-scream.md
@@ -1,0 +1,5 @@
+---
+'@capacitor-firebase/app-check': minor
+---
+
+feat(android): update Firebase App Check to 17.0.1

--- a/packages/app-check/README.md
+++ b/packages/app-check/README.md
@@ -19,8 +19,8 @@ See [Set up your Firebase project](https://firebase.google.com/docs/app-check/an
 
 This plugin will use the following project variables (defined in your app’s `variables.gradle` file):
 
-- `$firebaseAppCheckPlayIntegrityVersion` version of `com.google.firebase:firebase-appcheck-playintegrity` (default: `16.1.2`)
-- `$firebaseAppCheckDebugVersion` version of `com.google.firebase:firebase-appcheck-debug` (default: `16.1.2`)
+- `$firebaseAppCheckPlayIntegrityVersion` version of `com.google.firebase:firebase-appcheck-playintegrity` (default: `17.0.1`)
+- `$firebaseAppCheckDebugVersion` version of `com.google.firebase:firebase-appcheck-debug` (default: `17.0.1`)
 
 ### iOS
 

--- a/packages/app-check/android/build.gradle
+++ b/packages/app-check/android/build.gradle
@@ -3,8 +3,8 @@ ext {
     androidxAppCompatVersion = project.hasProperty('androidxAppCompatVersion') ? rootProject.ext.androidxAppCompatVersion : '1.6.1'
     androidxJunitVersion = project.hasProperty('androidxJunitVersion') ? rootProject.ext.androidxJunitVersion : '1.1.5'
     androidxEspressoCoreVersion = project.hasProperty('androidxEspressoCoreVersion') ? rootProject.ext.androidxEspressoCoreVersion : '3.5.1'
-    firebaseAppCheckPlayIntegrityVersion = project.hasProperty('firebaseAppCheckPlayIntegrityVersion') ? rootProject.ext.firebaseAppCheckPlayIntegrityVersion : '16.1.2'
-    firebaseAppCheckDebugVersion = project.hasProperty('firebaseAppCheckDebugVersion') ? rootProject.ext.firebaseAppCheckDebugVersion : '16.1.2'
+    firebaseAppCheckPlayIntegrityVersion = project.hasProperty('firebaseAppCheckPlayIntegrityVersion') ? rootProject.ext.firebaseAppCheckPlayIntegrityVersion : '17.0.1'
+    firebaseAppCheckDebugVersion = project.hasProperty('firebaseAppCheckDebugVersion') ? rootProject.ext.firebaseAppCheckDebugVersion : '17.0.1'
 }
 
 buildscript {

